src/orquestra/quantum/circuits/_gates.py

```diff
@@ -373,6 +376,11 @@ def exp(self) -> "Gate":
     def power(self, exponent: float) -> "Gate":
         return Power(self, exponent)
 
+    def __str__(self):
```

Comment:
This logic smells vulnerable to me. 
What if we have an exponent of the dagger, of controlled gate of the dagger of a gate?
Would this still work in some crazy, layered-wrapped scenarios like this?
I don't say it won't, but could you add a couple of more "crazy tests" to account for cases like this?

Reply:
Yes, it should. But I'll add some tests anyways. 😄